### PR TITLE
APPS-2968: library also should update annotation of method that updat…

### PR DESCRIPTION
**Pull request by olehnovatskyi-spryker**
- Developer(s): @oleh.novatskyi 

- Ticket: [APPS-2968](https://spryker.atlassian.net/browse/APPS-2968)

- merge: merge

- release: https://release.spryker.com/release/package?repo=spryker%2Fpropel-encryption-behavior&pr=3

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   PropelEncryptionBehavior | patch                 |                       |

-----------------------------------------

##### Change log

Fixes

- Adjusted `EncryptionBehavior::addEncryptionToSetter()` so it updates annotation for for setter method of the column with `BLOB` type. Now they receive `string` instead of `resource`.
- Adjusted `EncryptionBehavior::addDecryptionToGetter()` so it updates annotation for for getter method of the column with `BLOB` type. Now they return `string` instead of `resource`.

### CODE REVIEW

File: src/EncryptionBehavior.php [hunk 1]
```diff
@@ -330,22 +332,24 @@ protected static function insertEncryptedColumnsDeclaration(string &$script, int
...
+    protected function addEncryptionToSetter(
+        string &$script,
+        string $columnPhpName,
+        bool $isSearchable,
+        bool $hasColumnBlobType
+    ): void {
+        $setterLocation = $this->getMethodLocation($script, "set$columnPhpName");
+
+        if ($hasColumnBlobType) {
+            $paramAnnotationLocation = strpos($script, 'param resource', $setterLocation - 300);
```
Comment (ghost) [inline src/EncryptionBehavior.php]: Perhaps we should implement a method that finds the nearest previous '}' before the name of the method, in order to accurately determine the phpdoc in which wehave to make a replacement.

File: src/EncryptionBehavior.php [hunk 2]
```diff
@@ -407,4 +415,28 @@ protected function addDecryptionToGetter(string &$script, string $columnPhpName)
...
+    /**
+     * @param string $script
+     * @param string $methodName
+     *
+     * @throws \Exception
+     *
+     * @return int
+     */
+    protected function getMethodLocation(
+        string &$script,
```
Comment (dereuromark) [inline src/EncryptionBehavior.php]: Why &?
If possible we should not pass vars this way.

Comment (oleksander-kiiashko) [inline src/EncryptionBehavior.php]: Why do we need argument &$script, not just $script?

**Review by oleksander-kiiashko — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: src/EncryptionBehavior.php [hunk 3]
```diff
@@ -98,9 +99,10 @@ public function objectFilter(string &$script): void
             }
 
             $columnPhpName = $column->getPhpName();
+            $hasColumnBlobType = $column->getType() === PropelTypes::BLOB;
```
Comment (stereomon) [inline src/EncryptionBehavior.php]: IMO it should be `$isBlobTypeColumn`.

**Review by stereomon — CHANGES_REQUESTED** [review]
We should have tests for this. Can you please add some?

File: src/EncryptionBehavior.php (hunk 3 above)
Comment (dereuromark) [reply src/EncryptionBehavior.php]: Please reread the suggestion

Comment (dereuromark) [reply src/EncryptionBehavior.php]: Mainly about not using Yoga speak:)

**Comment by dereuromark** [comment]
If this is supposed to be released, we should also see a release template + changelog as per normal guidelines.

**Comment by olehnovatskyi-spryker** [comment]
> If this is supposed to be released, we should also see a release template + changelog as per normal guidelines.

Ok, I will prepare it.

**Review by stereomon — APPROVED** [review]
Code wise ok for me now. Please add the release template as mentioned by @dereuromark

**Comment by olehnovatskyi-spryker** [comment]
@dereuromark I added the template and changelogs. Check it, please.

**Comment by dereuromark** [comment]
Almost, please use code fencing instead of bold for Code pieces etc.

**Comment by olehnovatskyi-spryker** [comment]
Done

**Comment by olehnovatskyi-spryker** [comment]
Are we going to have a new release tag?

**Comment by dereuromark** [comment]
Done.
For the future it helps if you add the release link (with template check for validity) in the PR template for standalone packages (not going through RG release process).

**Pull request merged**